File: informatika ismeretek/kozep/2020_minta/Egyszamjatek.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Mivaaa {
    public class Egyszamjatek {
        
        public static void Main() {
            var jatekosok = Beolvas("egyszamjatek1.txt");
            
            Console.WriteLine("3. Feladat: Játékosok száma: " + jatekosok.Count);
            Console.WriteLine("4. Feladat: Írj be egy forduló számot!");
            
            var beForduloSzam = int.Parse(Console.ReadLine()) - 1;
            var forduloTotal = 0F;
            
            foreach(var tippek in jatekosok.Values) {
                forduloTotal += tippek[beForduloSzam];
            }
            
            Console.WriteLine("5. Feladat: Tippek átlaga: " + (forduloTotal / jatekosok.Count).ToString("0.00"));
            Console.Read();
        }

        public static Dictionary<string, int[]> Beolvas(string file){
            var jatekosok = new Dictionary<string, int[]>();
        
            foreach(var line in File.ReadAllLines(file)) {
                var split = line.Split(' ');
                var tippek = new int[split.Length - 1];
                
                for(var i = 1; i < split.Length; ++i) {
                    tippek[i - 1] = int.Parse(split[i]);
                }
            
                jatekosok.Add(split[0], tippek);
            }

            return jatekosok;
        }
    }
}

File: informatika ismeretek/kozep/2020_minta/java/Egyszamjatek.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class Egyszamjatek {

    @SuppressWarnings("boxing")
    public static void main(String[] args) throws IOException {
        var jatekosok = beolvas("egyszamjatek1.txt");
        
        System.out.println("3. Feladat: Játékosok száma: " + jatekosok.size());
        System.out.println("4. Feladat: Írj be egy forduló számot!");
        
        try(var conIn = new Scanner(System.in)){
            var beForduloSzam = conIn.nextInt() - 1;
            var forduloTotal = 0F;
            
            for(var tippek : jatekosok.values()) {
                forduloTotal += tippek[beForduloSzam];
            }
            
            System.out.printf("5. Feladat: Tippek átlaga: %.2f", forduloTotal / jatekosok.size());
        }
    }
    
    public static HashMap<String, int[]> beolvas(String file) throws IOException{
        var jatekosok = new HashMap<String, int[]>();
        
        for(var line : Files.readAllLines(Path.of(file))) {
            var split = line.split(" ");
            var tippek = new int[split.length - 1];
            
            for(var i = 1; i < split.length; ++i) {
                tippek[i - 1] = Integer.parseInt(split[i]);
            }
            
            jatekosok.put(split[0], tippek);
        }
        
        return jatekosok;
    }
}